actions/GenericAction.js

```diff
@@ -24,7 +24,9 @@ function createResponse(status, body) {
 }
 
 function createSuccessResponse(body) {
-  return createResponse(200, body);
+  // If there is a status code in the body, use it; otherwise, default to 200.
+  const { statusCode, ...response } = body;
+  return createResponse(statusCode ?? 200, response);
```

Comment:
This code is causing issues when loading target audience from `Adobe Target` and `CSV file` as they have dependency with `GenericAction.js`.

Reply:
@askayastha22 , I will check that, thank you

Reply:
@askayastha22 , I initially assumed the `body` would always be an object, but it seems that's not the case; different actions can produce different responses. <-- That is where TS shines.

@askayastha22 , @vtsaplin , I propose creating a tech debt ticket to standardize action responses. Something like
```
{
  statusCode: "integer",
  result: "object"
}
```
What do you think?

For now, I'll also check if the body is an object to avoid breaking the **existing** functionality.

Reply:
This is the relevant commit https://github.com/adobe/aem-genai-assistant/pull/261/commits/d4b0fb7e0b30778381ec754786b00a604cda1d47